File: src/main/java/com/github/trcdeveloppers/clayium/interfaces/IConnectable.java
package com.github.trcdeveloppers.clayium.interfaces;

import net.minecraft.util.EnumFacing;

public interface IConnectable {
    default int getInput(EnumFacing facing) {
        return -1;
    }

    default int getOutput(EnumFacing facing) {
        return -1;
    }

    default int rollInput(EnumFacing facing) {
        return -1;
    }

    default int rollOutput(EnumFacing facing) {
        return -1;
    }

    void setInput(EnumFacing facing, int mode);

    void setOutput(EnumFacing facing, int mode);

    void clearInput(EnumFacing facing);

    void clearOutput(EnumFacing facing);

}

File: src/main/kotlin/com/github/trcdeveloppers/clayium/common/interfaces/IConnectable.kt
package com.github.trcdeveloppers.clayium.common.interfaces

import net.minecraft.util.EnumFacing

interface IConnectable {
    fun getInput(facing: EnumFacing): Int {
        return -1
    }

    fun getOutput(facing: EnumFacing): Int {
        return -1
    }

    fun rollInput(facing: EnumFacing): Int {
        return -1
    }

    fun rollOutput(facing: EnumFacing): Int {
        return -1
    }

    fun setInput(facing: EnumFacing, mode: Int)
    fun setOutput(facing: EnumFacing, mode: Int)
    fun clearInput(facing: EnumFacing)
    fun clearOutput(facing: EnumFacing)
}
